src/socket_base.cpp

```diff
@@ -1792,18 +1834,22 @@ void zmq::socket_base_t::monitor_event (
                 memcpy (zmq_msg_data (&msg), &event_, sizeof event_);
                 zmq_msg_send (&msg, _monitor_socket, ZMQ_SNDMORE);
 
-                //  Send value in second frame (64bit unsigned)
-                zmq_msg_init_size (&msg, sizeof value_);
-                memcpy (zmq_msg_data (&msg), &value_, sizeof value_);
-                zmq_msg_send (&msg, _monitor_socket, ZMQ_SNDMORE);
+                //  Send values in second-Nth frames (64bit unsigned)
```

Comment:
I would suggest to first send a message part containing the number of values that will follow.

Reply:
yeah, makes sense